ts/components/services/ServicesSearch.tsx

```diff
@@ -33,47 +29,48 @@ type State = {
 /**
  * Filter only the services that match the searchText.
  */
-const generateSectionsServicesStateMatchingSearchTextArrayAsync = (
-  potServicesState: pot.Pot<
-    // tslint:disable-next-line: readonly-array
-    Array<SectionListData<pot.Pot<ServicePublic, Error>>>,
-    Error
-  >,
+const generateSectionsServicesStateMatchingSearchTextArray = (
+  // tslint:disable-next-line: readonly-array
+  servicesState: Array<SectionListData<pot.Pot<ServicePublic, Error>>>,
   searchText: string
   // tslint:disable-next-line: readonly-array
-): Promise<Array<SectionListData<pot.Pot<ServicePublic, Error>>>> => {
-  return new Promise(resolve => {
-    const result = pot.getOrElse(
-      pot.map(potServicesState, _ =>
-        _.filter(sectionList =>
-          sectionDataContainsText(sectionList, searchText)
-        )
-      ),
-      []
+): Array<SectionListData<pot.Pot<ServicePublic, Error>>> => {
+  // tslint:disable-next-line: readonly-array prefer-const no-var-keyword
+  var result: Array<SectionListData<pot.Pot<ServicePublic, Error>>> = [];
+  servicesState.forEach(sectionList => {
```

Comment:
use map + [flatten](https://lodash.com/docs/4.17.11#flatten)

Reply:
In the PR the code for filtering is different. Can you check if that's okay? thanks